### PR TITLE
chore: replace tsdx with tsup and make ts-sdk dual module

### PR DESCRIPTION
**Pull request by jjleng**
### Description
tsdx did not work well for ts-sdk bundling. Due to the dependencies that tsdx uses, spread operator of Uint8Array doesn't work. Also the generated esm module had some issues.

This PR makes ts-sdk a true dual module. SDK can now be used in both esm and cjs apps. PR also adds back `form-data` as a dependency, which is required by the generated code of openapi client.

esm apps need to add `type: module` in their `package.json` to import the SDK as an esm module.

### Test Plan
Manually build and tested

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3918)

### CODE REVIEW

File: ecosystem/typescript/sdk/src/transaction_builder/builder.ts [hunk 1]
```diff
@@ -33,6 +33,9 @@ import * as Gen from "../generated/index";
 
 export { TypeTagParser } from "./builder_utils.js";
 
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (banool) [inline ecosystem/typescript/sdk/src/transaction_builder/builder.ts:36]: Can we just name it according to the convention to satisfy this lint?

**Review by banool — APPROVED** [review]
Can you check that this makes it possible to remove the `tslib` dep from the examples? It is very wack that the JS examples require it, I'm hoping this fixes the problem.

File: ecosystem/typescript/sdk/src/transaction_builder/builder.ts (hunk 1 above)
Comment (jjleng) [reply ecosystem/typescript/sdk/src/transaction_builder/builder.ts]: done

**Comment by jjleng** [comment]
`tslib` is not needed to run the examples

**Review by gregnazario — APPROVED** [review]

**Pull request merged**